### ISSUE TITLE
Version Changes ?

### ISSUE THREAD

**Issue by MSP-Greg** [rails/rails#28231]
Several files in the stable branch have previously changed with new releases.  For the most recent change, they did not. See:

[version.rb](https://github.com/rails/rails/commits/5-0-stable/version.rb)
[RAILS_VERSION](https://github.com/rails/rails/commits/5-0-stable/RAILS_VERSION)
[railties/lib/rails/gem_version.rb](https://github.com/rails/rails/commits/5-0-stable/railties/lib/rails/gem_version.rb)
[rails/activemodel/lib/active_model/gem_version.rb](https://github.com/rails/rails/commits/5-0-stable/activemodel/lib/active_model/gem_version.rb)

Also, there is now a branch for the current release ([5.0.2.rc1](https://github.com/rails/rails/tree/5-0-2)), which seems a little odd, as they're normally tagged, with all their commits in the stable branch...

**Comment by eugeneius** [comment]
The process of using a release branch is documented [here](https://github.com/rails/rails/blob/master/RELEASING_RAILS.md#create-a-release-branch):

> From the stable branch, create a release branch. For example, if you're releasing Rails 3.0.10, do this:
>
>     [aaron@higgins rails (3-0-stable)]$ git checkout -b 3-0-10
>     Switched to a new branch '3-0-10'
>     [aaron@higgins rails (3-0-10)]$

The release branch will be merged back into the stable branch once the final release is out. Here's where that happened for 5.0.1: https://github.com/rails/rails/commit/bed94202de9da55e35475f9c2ada66a4ef9e2491

**Comment by MSP-Greg** [comment]
@eugeneius 
Okay, I've read 'Releasing Rails'.  Seems very odd that the stable branch does not reflect the version that it currently contains.

Looking at it another way, with the start of the release of 5.0.2, there will not be any further changes to 5.0.1. Hence, why is the stable branch at version 5.0.1?  Not looking for an answer...

**Comment by rafaelfranca** [comment]
The release candidates are cut from a release branch and only after the final release is made is that the commits are merged back to the stable branch. So yes, while the final is not released the stable branch will be one release back from the last release in that line. This is to make possible to continue development in the stable branches without blocking the contributors.

**Comment by MSP-Greg** [comment]
@rafaelfranca Thanks for the reply.

Yes, I get all of that.  But during the 'in between' time of a stable update release (eg 5.0.*) --

1. Since Rails seems to have a lot of docs re release, etc, where is it described what type commits should go in 'stable' vs 'release branch' (eg 5-0-2)?

2. I don't see why this structure / methodology is affected by the version info contained in `version.rb` and `RAILS_VERSION`.

As you know, the stable branch normally shows version info that is the last released version, and, hence, all commits will be in the next release.  I don't see why that needs to change while a release is in progress.

Regardless, not looking for a response, and thanks for all your effort.

BTW, the majority of hits on my doc site are to rails master.  I'm debating whether to add docs for tests...

**Comment by rafaelfranca** [comment]
1. After the release branch is cut, only bug fixes for regressions in the release are commited to that branch. After the final release is made the release branch is deleted and the stable branch will contain all the changes of the release branch.

2. The version files are only updated in the release branch. To avoid overhead of syncing the release branch and the stable branch, we only change the version in the release branch. It is just a few days and we are totally fine with this inconsistency. No users are affected by it.
